src/spyglass/utils/dj_mixin.py

```diff
@@ -137,6 +137,27 @@ def _safe_context(cls):
             else nullcontext()
         )
 
+    @classmethod
+    def get_fully_defined_key(
+        cls, key: dict = None, required_fields: list[str] = None
+    ) -> dict:
+        if key is None:
+            key = dict()
+
+        required_fields = required_fields or cls.primary_key
+        if isinstance(key, (str, dict)):  # check is either keys or substrings
+            if not all(
+                field in key for field in required_fields
+            ):  # check if all required fields are in key
+                if not len(query := cls() & key) == 1:  # check if key is unique
+                    raise KeyError(
+                        f"Key {key} is neither fully specified nor a unique entry in"
+                        + f"{cls.full_table_name}"
```

Comment:
I think people might wonder what they should have specified to get this error. 

I try to structure error messages to put the variables at the end of the message. If they're super long, it can be tough to parse the sentence: `This thing ...wall of text ... is whatever` vs `Thing is whatever: wall of text`

```suggestion
                        f"Key is neither fully specified nor a unique entry in"
                        + f"table.\n\tTable: {cls.full_table_name}\n\tKey: {key}"
                        + f"Required fields: {required_fields}\n\tResult: {query}"
```